controllers/secretagentconfiguration_controller.go

```diff
@@ -109,7 +151,19 @@ func (reconciler *SecretAgentConfigurationReconciler) Reconcile(req ctrl.Request
 					"secret_name", secretReq.Name,
 					"data_key", key.Name,
 					"secret_type", string(key.Type))
-				keyInterface.LoadSecretFromManager(ctx, &instance.Spec.AppConfig, instance.Namespace, secretReq.Name)
+				if err := keyInterface.LoadSecretFromManager(ctx, &instance.Spec.AppConfig, instance.Namespace, secretReq.Name); err != nil {
+					// TODO We should fail/retry. Temporary hack until keytool + trustore secretManager is implemented
```

Comment:
as it stands, yes. Because if `useSecMgr` is true, we don't even attempt to load from local k8s secrets. The operator goes into an infinite loop creating keytool and truststores. . This TODO will be removed shortly